plugins/inputs/sqlserver/README.md

```diff
@@ -52,6 +52,10 @@ GO
     "Server=192.168.1.10;Port=1433;User Id=<user>;Password=<pw>;app name=telegraf;log=1;",
   ]
 
+  ## Authentication method
+  ## valid methods: "default", "AAD"
+  # auth_method = "default"
```

Comment:
Should we name it `"credentials"` or `"connection param"`? `"default"` is not very descriptive. :-)

Reply:
@akrantz01  and I agree, we think "credentials" would be more descriptive.

Reply:
I switched to "connection_string". I considered using that initially but I thought it was too long. Also I thought it would be confusing for #9362 because in that case, the username/password is intentionally left out of the connection string to tell the client to use Windows Integrated auth. So to use WIA you would say auth_method = "connection_string", then not put auth in the connection string.

I don't know what's best here. "default" is not descriptive but not incorrect. "connection_string" may be the next best option.

Reply:
Sorry, but I don't like this; missing credentials should not be assumed...
The convention as far as I understand it for Windows Integrated Auth is to specify 'Integrated Security=true' in the connection string e.g. 
"Server=localhost;Port=1433;Integrated Security=true;log=1;"

Reply:
Hi @aslgithub, the issue here isn't personal preference or convention. If I could have my preference I would require an explicit setting to enable windows integrated security like you suggest, either in the connection string or as an option in a telegraf setting like "auth_method".

The issue is backward compatibility. Some existing users of this plugin (pre 1.19.0) use windows integrated auth without the "Integrated Security=true" connection parameter and without credentials. If sql server and the golang driver allow you to use windows integrated auth by omitting credentials, and telegraf allowed you to do it that way (and there are telegraf users that did it that way), we need to support it. We can't break the config for these users and force them to debug why their connection no longer works after they upgrade.

The design decision to try integrated security if there are no credentials in the connection string was made in the past, I assume by the SQL Server team, possibly many years ago. It never was our choice to do it this way.

Reply:
@reimda Aren't you now going to break this anyway by forcing all users to set an auth method or have I misunderstood the intention ?

Reply:
The default setting for auth_method will be to act like pre-1.19.0 telegraf ("connection_string"). Users who want to use Azure Active Directory will need to opt in by setting auth_method to "AAD". This way people upgrading to 1.19.1 from any version before 1.19.0 will not be affected at all.